fastapi_azure_auth/user.py

```diff
@@ -231,6 +231,16 @@ class Claims(BaseModel):
         description='The primary username that represents the user. Only available in V2.0 tokens',
     )
 
+    @validator('scp', pre=True)
+    def scopes_to_list(cls, v: object) -> object:
```

Comment:
In practice, it could receive something else than a `str`. One could build the User object directly (let's say to write a test case) passing it a list of string. That's why we want to have this `isinstance` check. If I type the method `str -> list[str]` then mypy complains that the second return is not reachable.

I however, agree that the typing is not pretty. I've skimmed the Pydantic doc, but I couldn't find examples of typed validator.

Reply:
No worries, if my pay complains it is what it is. Thank you 😊